packages/salesforcedx-vscode-core/src/settings/sfdxCoreSettings.ts

```diff
@@ -85,6 +86,14 @@ export class SfdxCoreSettings {
     return this.getConfigValue(FUNCTIONS_ENABLED, false);
   }
 
+  public async updateFunctionsPullDependencies(value: boolean) {
+    await this.setConfigValue(FUNCTIONS_PULL_DEPENDENCIES, value);
+  }
+
+  public getFunctionsPullDependencies(): boolean {
+    return this.getConfigValue(FUNCTIONS_PULL_DEPENDENCIES, true);
```

Comment:
Not in the scope of this PR - but the abstractions this file provides seem to be very limited.